File: js/audio_test.js
#!/usr/bin/env node
/**
 * Gemini 2.5 Flash Audio Understanding Test
 *
 * Tests basic audio processing capabilities:
 * - Transcription
 * - Description
 * - Analysis
 */

import { GoogleGenAI } from '@google/genai';
import { readFileSync } from 'fs';
import { basename, extname } from 'path';

const MIME_TYPES = {
  '.mp3': 'audio/mp3',
  '.wav': 'audio/wav',
  '.aiff': 'audio/aiff',
  '.aac': 'audio/aac',
  '.ogg': 'audio/ogg',
  '.flac': 'audio/flac',
};

async function main() {
  // Check for API key
  const apiKey = process.env.GOOGLE_API_KEY;
  if (!apiKey) {
    console.error('Error: GOOGLE_API_KEY environment variable not set');
    console.error('Get your key at: https://aistudio.google.com/apikey');
    process.exit(1);
  }

  // Check for audio file argument
  const audioPath = process.argv[2];
  if (!audioPath) {
    console.log('Usage: node audio_test.js <audio_file>');
    console.log('Supported formats: WAV, MP3, AIFF, AAC, OGG, FLAC');
    process.exit(1);
  }

  // Determine MIME type
  const ext = extname(audioPath).toLowerCase();
  const mimeType = MIME_TYPES[ext];
  if (!mimeType) {
    console.error(`Error: Unsupported audio format: ${ext}`);
    console.error('Supported: WAV, MP3, AIFF, AAC, OGG, FLAC');
    process.exit(1);
  }

  console.log(`Processing: ${audioPath}`);
  console.log('-'.repeat(50));

  // Initialize client
  const ai = new GoogleGenAI({ apiKey });

  // Upload the audio file
  console.log('Uploading audio file...');
  const audioData = readFileSync(audioPath);
  const base64Audio = audioData.toString('base64');

  const audioPart = {
    inlineData: {
      mimeType,
      data: base64Audio,
    },
  };

  // Test 1: Transcription
  console.log('\n[1] Transcription:');
  console.log('-'.repeat(30));
  let response = await ai.models.generateContent({
    model: 'gemini-2.5-flash',
    contents: [
      {
        role: 'user',
        parts: [
          { text: 'Transcribe this audio. Provide the full text.' },
          audioPart,
        ],
      },
    ],
  });
  console.log(response.text);

  // Test 2: Description
  console.log('\n[2] Audio Description:');
  console.log('-'.repeat(30));
  response = await ai.models.generateContent({
    model: 'gemini-2.5-flash',
    contents: [
      {
        role: 'user',
        parts: [
          { text: 'Describe this audio clip. What do you hear? Include details about speakers, tone, background sounds, etc.' },
          audioPart,
        ],
      },
    ],
  });
  console.log(response.text);

  // Test 3: Analysis
  console.log('\n[3] Content Analysis:');
  console.log('-'.repeat(30));
  response = await ai.models.generateContent({
    model: 'gemini-2.5-flash',
    contents: [
      {
        role: 'user',
        parts: [
          { text: 'Analyze this audio content. Summarize the main topics, identify any speakers, and note the overall sentiment.' },
          audioPart,
        ],
      },
    ],
  });
  console.log(response.text);

  console.log('\n' + '-'.repeat(50));
  console.log('Done!');
}

main().catch(console.error);

File: python/audio_test.py
#!/usr/bin/env python3
"""
Gemini 2.5 Flash Audio Understanding Test

Tests basic audio processing capabilities:
- Transcription
- Description
- Analysis
"""

import sys
import os
from pathlib import Path

from google import genai

def main():
    # Check for API key
    api_key = os.environ.get("GOOGLE_API_KEY")
    if not api_key:
        print("Error: GOOGLE_API_KEY environment variable not set")
        print("Get your key at: https://aistudio.google.com/apikey")
        sys.exit(1)

    # Initialize client
    client = genai.Client(api_key=api_key)

    # Check for audio file argument
    if len(sys.argv) < 2:
        print("Usage: python audio_test.py <audio_file>")
        print("Supported formats: WAV, MP3, AIFF, AAC, OGG, FLAC")
        sys.exit(1)

    audio_path = Path(sys.argv[1])
    if not audio_path.exists():
        print(f"Error: File not found: {audio_path}")
        sys.exit(1)

    print(f"Processing: {audio_path}")
    print("-" * 50)

    # Upload the audio file
    print("Uploading audio file...")
    uploaded_file = client.files.upload(file=str(audio_path))
    print(f"Uploaded: {uploaded_file.name}")

    # Test 1: Transcription
    print("\n[1] Transcription:")
    print("-" * 30)
    response = client.models.generate_content(
        model="gemini-2.5-flash",
        contents=[
            "Transcribe this audio. Provide the full text.",
            uploaded_file
        ]
    )
    print(response.text)

    # Test 2: Description
    print("\n[2] Audio Description:")
    print("-" * 30)
    response = client.models.generate_content(
        model="gemini-2.5-flash",
        contents=[
            "Describe this audio clip. What do you hear? Include details about speakers, tone, background sounds, etc.",
            uploaded_file
        ]
    )
    print(response.text)

    # Test 3: Analysis
    print("\n[3] Content Analysis:")
    print("-" * 30)
    response = client.models.generate_content(
        model="gemini-2.5-flash",
        contents=[
            "Analyze this audio content. Summarize the main topics, identify any speakers, and note the overall sentiment.",
            uploaded_file
        ]
    )
    print(response.text)

    # Cleanup
    print("\n" + "-" * 50)
    print("Done!")


if __name__ == "__main__":
    main()
